app/src/main/java/com/example/happybirthday/MainActivity.kt

```diff
@@ -54,44 +54,39 @@ class MainActivity : ComponentActivity() {
 }
 
 @Composable
-fun BirthdayGreetingWithText(message: String, from: String) {
+fun BirthdayGreetingWithText(message: String, from: String, modifier: Modifier = Modifier) {
     // Create a column so that texts don't overlap
-    Column {
-        // Add two text, let them appear on top of each other
+    Column(
+        modifier = modifier.fillMaxSize(),
+        verticalArrangement = Arrangement.Top,
+        horizontalAlignment = Alignment.CenterHorizontally
+    ) {
         Text(
             text = message,
             fontSize = 36.sp,
             modifier = Modifier
-                .fillMaxWidth()
-                .wrapContentWidth(Alignment.CenterHorizontally)
-                .padding(start = 16.dp, top = 16.dp)
-
+                .padding(top = 16.dp)
         )
         Text(
             text = from,
             fontSize = 24.sp,
             modifier = Modifier
-                .fillMaxWidth()
-                .wrapContentWidth(Alignment.CenterHorizontally)
-                .padding(start = 16.dp, end = 16.dp)
+                .padding(top = 16.dp)
         )
     }
 }
 
 @Composable
-fun BirthdayGreetingWithImage(message: String, from: String) {
-    val image = painterResource(R.drawable.androidparty)
+fun BirthdayGreetingWithImage(message: String, from: String, modifier: Modifier = Modifier) {
+    val image = painterResource(id = drawable.androidparty)
     // Create a box to overlap image and texts
     Box {
         Image(
             painter = image,
             contentDescription = null,
-            modifier = Modifier
-                .fillMaxHeight()
-                .fillMaxWidth(),
             contentScale = ContentScale.Crop
         )
-        BirthdayGreetingWithText(message = message, from = from)
+        BirthdayGreetingWithText(message = message, from = from, modifier)
```

Comment:
nit: do you need named parameters here? if you do want to keep them in, should we add it for modifier too?